examples/covasim_/vaccinating_elderly/example_vaccine.py

```diff
@@ -1,38 +1,34 @@
 # -*- coding: utf-8 -*-
+import os
+import logging
+import warnings
 import pandas as pd
-import numpy as np
-import covasim as cv  # Version used in our study is 3.07
-import random
 from causal_testing.specification.causal_dag import CausalDAG
 from causal_testing.specification.scenario import Scenario
 from causal_testing.specification.variable import Input, Output
 from causal_testing.specification.causal_specification import CausalSpecification
-from causal_testing.data_collection.data_collector import ExperimentalDataCollector
+from causal_testing.data_collection.data_collector import ObservationalDataCollector
 from causal_testing.testing.causal_test_case import CausalTestCase
 from causal_testing.testing.causal_test_outcome import Positive, Negative, NoEffect
 from causal_testing.testing.estimators import LinearRegressionEstimator
 from causal_testing.testing.base_test_case import BaseTestCase
 
-import os
-import logging
+warnings.filterwarnings("ignore", message=".*The 'nopython' keyword.*")  # Supress the numba warning in covasim
```

Comment:
Is this needed now covasim is removed?

Reply:
Thanks for your suggestions @cwild-UoS - I'll review them this morning. 

For some reason, the Numba warnings are still outputed even though covasim isn't directly called in this script. I wonder whether they're called indirectly through another module? But it does seem to be needed to supress the warnings.

Reply:
Have you run the script in an environment that doesn't have Covasim installed? Just curious if that would change anything

Reply:
@cwild-UoS I traced it back to EconML - the two modules that use Numba are Shap and Sparse, which are both implicitly installed. Do we need EconML? I can't find any usage of it in the framework

Reply:
I can't recall what EconML was used for off the top of my head :(. Might be worth going back to an earlier version and seeing where it was used and when it stopped being used? 

If there definitely are no uses then I can't see why it can't also be removed.

Reply:
@cwild-UoS It was only used in the `CausalForestEstimator`. This also ties in with the Python 3.11 branch you were working on, as you removed EconML from the .toml file in your commits